src/engine/enginebuffer.cpp

```diff
@@ -562,6 +562,8 @@ void EngineBuffer::ejectTrack() {
     doSeekFractional(0.0, SEEK_EXACT);
     m_pause.unlock();
 
+    m_pReader->newTrack(m_pCurrentTrack);
```

Comment:
```TrackPointer()``` might be more readable. 

Could you add a comment that it's to clean up state in CachingReader?